pkgs/applications/virtualization/qemu/default.nix

```diff
@@ -283,6 +283,7 @@ stdenv.mkDerivation rec {
     homepage = "http://www.qemu.org/";
     description = "A generic and open source machine emulator and virtualizer";
     license = licenses.gpl2Plus;
+    mainProgram = "qemu-kvm";
```

Comment:
Isn't `qemu-kvm` only available on Linux? Not sure if this is the best default binary.

Reply:
> Isn't `qemu-kvm` only available on Linux? Not sure if this is the best default binary.

Have no idea.

I see this:

https://github.com/NixOS/nixpkgs/blob/57e660a35eccd602e8d1811df7b8eb0701b8585d/pkgs/applications/virtualization/qemu/default.nix#L244-L253

Reply:
I think using `meta.mainProgram = "qemu-system-${stdenv.hostPlatform.qemuArch}"` a better approach them.

Reply:
What is the most correct thing to be done? 

Would it make sense use `qemu-kvm` only when in linux and conditionally check that to build the string "qemu-kvm" and even other "names" for non linux? :thinking:

Reply:
Actually, `qemu-kvm` seems fine since it will always exist.

Reply:
> I think using `meta.mainProgram = "qemu-system-${stdenv.hostPlatform.qemuArch}"` a better approach them.

I am going to test it.

Reply:
I don't think you need to bother. This `qemu-kvm` wrapper seems smart enough that if `kvm` is not available it will not use it.